From 2099ce737815502769fd738f680721b0d8a6ecae Mon Sep 17 00:00:00 2001
From: Jason Chen CJ <jason.cj.chen@intel.com>
Date: Tue, 23 Jan 2018 23:03:19 +0800
Subject: [PATCH 0509/1240] Check x86_hyper type before doing hypercall

this is to fix native boot failure issue with CONFIG_CWP

Change-Id: I735283cbf462c8b79d9742d64950685d6ae552c1
Signed-off-by: Jason Chen CJ <jason.cj.chen@intel.com>
---
 drivers/char/vhm/vhm_dev.c | 7 +++++++
 drivers/cwp/cwp_trace.c    | 7 +++++++
 2 files changed, 14 insertions(+)

diff --git a/drivers/char/vhm/vhm_dev.c b/drivers/char/vhm/vhm_dev.c
index ee9df32..d45953d 100644
--- a/drivers/char/vhm/vhm_dev.c
+++ b/drivers/char/vhm/vhm_dev.c
@@ -84,6 +84,8 @@
 #include <linux/vhm/vhm_vm_mngt.h>
 #include <linux/vhm/vhm_hypercall.h>
 
+#include <asm/hypervisor.h>
+
 #define  DEVICE_NAME "cwp_vhm"
 #define  CLASS_NAME  "vhm"
 
@@ -590,6 +592,11 @@ static int __init vhm_init(void)
 
 	pr_info("vhm: initializing\n");
 
+	if (x86_hyper_type != X86_HYPER_CWP) {
+		pr_err("vhm: not support cwp hypervisor!\n");
+		return -EINVAL;
+	}
+
 	if (hcall_get_api_version(virt_to_phys(&api_version)) < 0) {
 		pr_err("vhm: failed to get api version from Hypervisor !\n");
 		return -EINVAL;
diff --git a/drivers/cwp/cwp_trace.c b/drivers/cwp/cwp_trace.c
index 0f9e7f3..8f0fd8e 100644
--- a/drivers/cwp/cwp_trace.c
+++ b/drivers/cwp/cwp_trace.c
@@ -63,6 +63,8 @@
 #include <linux/fs.h>
 #include <linux/mm.h>
 
+#include <asm/hypervisor.h>
+
 #include "sbuf.h"
 
 
@@ -211,6 +213,11 @@ static int __init cwp_trace_init(void)
 	int ret = 0;
 	int i, cpu;
 
+	if (x86_hyper_type != X86_HYPER_CWP) {
+		pr_err("cwp_trace: not support cwp hypervisor!\n");
+		return -EINVAL;
+	}
+
 	/* TBD: we could get the native cpu number by hypercall later */
 	pr_info("%s, cpu_num %d\n", __func__, nr_cpus);
 	if (nr_cpus > MAX_NR_CPUS) {
-- 
2.7.4

